### PR TITLE
release(Apollo): Update to Apollo v3 and GraphQL v15 

### PR DESCRIPTION
**Pull request by nilgradisnik**
to: @williaster @alecklandgraf

## Description

Update Apollo client to v3 and GraphQL v15

## Motivation and Context

Apollo client [v3](https://www.apollographql.com/blog/announcing-the-release-of-apollo-client-3-0/) has been out for quite some time and we want to make sure Lunar is up to date. This release contains quite a few breaking changes so this PR is following the [migration guide](https://www.apollographql.com/docs/react/migrating/apollo-client-3-migration/) consolidating packages to `@apollo/client` and taking care of breaking changes.

This PR also updates GraphQL to [v15](https://github.com/graphql/graphql-js/releases/tag/v15.0.0) making sure we're up to date.

Also migrated to use `lodash/fp`.

## Testing

Updated unit tests. Tested in Storybook.

## Screenshots

N/A

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

File: packages/apollo/src/updaters/addToList.ts [hunk 1]
```diff
@@ -36,7 +33,7 @@ export default function addToList<Result, Vars = {}>(
       }
     }
 
-    set(nextResult, listPath, [...list, data]);
+    const nextResult = set(listPath, [...list, data], { ...queryResult });
```
Comment (nilgradisnik) [inline packages/apollo/src/updaters/addToList.ts:36]: `get` and `set` function from `lodash/fp` have different order of arguments

File: packages/apollo/src/updaters/removeFromList.ts [hunk 2]
```diff
@@ -1,22 +1,20 @@
...
+  docOrQuery: DocumentNode | DataProxy.Query<Vars, Result>,
   listPath: string,
   id: string | number,
   idName: string = 'id',
 ) {
-  const query = prepareQuery<Vars>(docOrQuery);
+  const query = prepareQuery<Result, Vars>(docOrQuery);
 
-  return (cache: DataProxy) => {
+  return (cache: ApolloCache<Result>) => {
```
Comment (nilgradisnik) [inline packages/apollo/src/updaters/removeFromList.ts:15]: `ApolloCache` class implements `DataProxy` interface and we need it to evict cache

File: packages/apollo/src/updaters/removeFromList.ts [hunk 3]
```diff
@@ -26,10 +24,19 @@ export default function removeFromList<Result, Vars = {}>(
...
-    set(
-      nextResult,
+    const rootItem = listPath.split('.')[0];
+    // @ts-ignore
+    const resultRoot = queryResult[rootItem] as StoreObject;
+    // Evict cache before writeQuery: https://github.com/apollographql/apollo-client/issues/6451
+    cache.evict({
+      id: cache.identify(resultRoot),
+      broadcast: false,
+    });
```
Comment (nilgradisnik) [inline packages/apollo/src/updaters/removeFromList.ts:34]: As stated here https://github.com/apollographql/apollo-client/issues/6451#issuecomment-645612063 we need to evict cache when removing items from a list

File: packages/apollo/src/updaters/removeFromList.ts [hunk 4]
```diff
@@ -26,10 +24,19 @@ export default function removeFromList<Result, Vars = {}>(
       }
     }
 
-    set(
-      nextResult,
+    const rootItem = listPath.split('.')[0];
+    // @ts-ignore
+    const resultRoot = queryResult[rootItem] as StoreObject;
```
Comment (nilgradisnik) [inline packages/apollo/src/updaters/removeFromList.ts:29]: Not sure if there's a better way to identify nested nodes but to evict cache we need at least the top level node ID for the `cache.identify` call below.

File: packages/apollo/test/updaters/__snapshots__/addToList.test.ts.snap [hunk 5]
```diff
@@ -4,13 +4,4 @@ exports[`addToList() errors if mutation data does not exist 1`] = `"Cannot find
...
-"Can't find field things on object {
-  \\"something\\": {
-    \\"type\\": \\"id\\",
-    \\"generated\\": false,
-    \\"id\\": \\"something:123\\",
-    \\"typename\\": \\"something\\"
-  }
-}."
-`;
+exports[`addToList() errors if query does not exist in cache 1`] = `"\\"otherQuery\\" list \\"something.things\\" is not an array."`;
```
Comment (nilgradisnik) [inline packages/apollo/test/updaters/__snapshots__/addToList.test.ts.snap:7]: `cache.readQuery` does not throw an error if there's no results anymore, it returns `null` instead. That's why the other error is thrown here.

**Review by alecklandgraf — APPROVED** [review]

**Comment by williaster** [comment]
Ummm, we should no longer have commits in this repo as it's deprecated and has been merged into pineapple. @alecklandgraf do you disagree?

**Comment by williaster** [comment]
If you still need to make changes to particular packages, we should maybe move those packages to be internal?

**Comment by alecklandgraf** [comment]
@williaster I think this patch migrates the Apollo client to the version in pineapple.

**Pull request merged**